### PR TITLE
__init__.py: Add description of coalib.

### PR DESCRIPTION
**Pull request by wimpykid26**
The existing help(coalib) command issues a list of packages inside coalib
without any description.Thus a small description of the coalib package
would help.

Fixes https://github.com/coala/coala/issues/1663

### CODE REVIEW

**Comment by sils1297** [comment]
if you run coala locally it'll tell you the very same things that gitmate tells you. This way you don't have to push to get the results, it can also autopatch some things

**Comment by wimpykid26** [comment]
@sils1297 my latest commit has removed that piece of code but the gitmate-bot is reviewing the earlier code. I have checked it locally on my system.

**Comment by sils1297** [comment]
Gitmate wants you to squash everything together, it reviews every single
commit. You can do this with git rebase -i master

**Comment by sils1297** [comment]
A rebase can help you squashing your commits togther. [This page](http://coala.io/git) may help you to get started on this. (*Do not press the update branch button. It's evil.*)

Add the original coala repository as a remote:

```
git remote add upstream https://github.com/coala/coala.git
```

Then go to your branch and rebase it onto our master:

```
git checkout <yourbranch>
git fetch upstream
git rebase --interactive upstream/master
```

In this editor which is now open you can use `fixup` to squash commits together - just read the instructions provided by git. If weird things happen, keep using `git status` to see whats going on and `git rebase --continue` to get to the next step of your rebase if needed.

Now push it up. You will need to do a force push because you change history here (do not do this on master branches :)):

```
git push --force
```

If you're just looking to fix an issue very quickly and not interested in contributing to coala long term, we can fix it up for you - just tell us :).

Ping us on [our gitter channel](https://coala.io/chat) if you need help!

File: coalib/__init__.py [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 
 def get_version():
+    '''The coalib package is a collection of various subpackages
```
Comment (sils1297) [inline coalib/__init__.py]: Can you use double quotes? Also a newline after the triple quotes and before the ending ones is what we do usually

File: coalib/__init__.py [hunk 2]
```diff
@@ -6,6 +6,10 @@
...
+       regarding writing and editing bears.
+       Various other packages such as formatting and
+       settings are also included in coalib.'''
```
Comment (sils1297) [inline coalib/__init__.py]: are you sure this is the right place btw.?

Comment (wimpykid26) [inline coalib/__init__.py]: @sils1297 This file runs when help(coalib) executes, I earlier used a print statement to do the task but later found out the description also shows when I use a bear to test a file. Therefore replaced the print statement with a docstring inside the function.

File: coalib/__init__.py (hunk 1 above)
Comment (wimpykid26) [inline coalib/__init__.py]: Double quotes will essentially lead to a print statement which again will execute even in other processes such as using a bear to test a script. Therefore, I used a docstring. Added a newline before and after the quotes.

Comment (wimpykid26) [inline coalib/__init__.py]: @sils1297 I hugely appreciate such a detailed explanation of the problem. I feel humbled. I squashed the commits into one as you told me so. As you suggested, to begin with I chose this newcomer issue but I have been following the tutorials and would love to contribute to coala in the long run. Thanks a lot.

File: coalib/__init__.py [hunk 3]
```diff
@@ -6,6 +6,12 @@
```
Comment (Adrianzatreanu) [inline coalib/__init__.py]: commit shortlog: __init.py__: Add help description

im pretty sure its __init__.py, not __init.py__ :D

File: coalib/__init__.py [hunk 4]
```diff
@@ -6,6 +6,12 @@
 
 
 def get_version():
+    '''
```
Comment (Adrianzatreanu) [inline coalib/__init__.py]: werent docstrings enclosed in """ and not ''' ?

Comment (wimpykid26) [inline coalib/__init__.py]: @Adrianzatreanu Enclosed them in """ in the new commit.

File: coalib/__init__.py (hunk 3 above)
Comment (wimpykid26) [inline coalib/__init__.py]: @Adrianzatreanu Fixed the typo, thanks for pointing that out :D

**Comment by wimpykid26** [comment]
@Adrianzatreanu The issue stated that there was no description of coalib in the help documentation. If I add the description outside the get_version function, it gets executed repeatedly while testing a file with a bear or applying settings to a bear( basically everytime __init__.py file runs). That is why I wrote it inside the get_version function.

**Comment by sils1297** [comment]
so if you do the following:

```
$ python
> import coalib
> help(coalib)
```

that help is shown?

**Comment by sils1297** [comment]
can we maybe have a test for this?

**Comment by wimpykid26** [comment]
Yes it does, but still inside the function get_version . How to write the test for that?

**Comment by sils1297** [comment]
I think you should look at http://helloworldbookblog.com/python-help-and-your-modules/ , IIUC the docstring has to be at module level for help to work correctly. I'll leave it to you to figure it out, ping me if I'm wrong.

**Comment by wimpykid26** [comment]
@sils1297 you are correct. Thanks a lot for the help. I relocated the docstring to the first line of the file and now , it seems to be working fine :D

File: coalib/__init__.py [hunk 5]
```diff
@@ -1,3 +1,9 @@
+"""
+       The coalib package is a collection of various subpackages
+       regarding writing and editing bears.
+       Various other packages such as formatting and
+       settings are also included in coalib.
+"""
```
Comment (sils1297) [inline coalib/__init__.py]: can we have some newlines here?

File: coalib/__init__.py [hunk 6]
```diff
@@ -1,3 +1,9 @@
...
+       The coalib package is a collection of various subpackages
+       regarding writing and editing bears.
+       Various other packages such as formatting and
+       settings are also included in coalib.
```
Comment (sils1297) [inline coalib/__init__.py]: why is this indented?

**Comment by sils1297** [comment]
also please squash your commits, we don't really want those "oh damn this wasn't like I wanted it" fixup commits in our history ;)

**Comment by sils1297** [comment]
(You can squash with rebase -i, ask someone on gitter for help if needed)

File: coalib/__init__.py (hunk 5 above)
Comment (wimpykid26) [inline coalib/__init__.py]: newlines as in empty ones ?

**Comment by sils1297** [comment]
yes

File: coalib/__init__.py (hunk 6 above)
Comment (wimpykid26) [inline coalib/__init__.py]: will remove the indentation.

**Comment by wimpykid26** [comment]
@sils1297 Please review

**Comment by sils1297** [comment]
quote from the failing CircleCI build:

```
Project wide:
|    | [NORMAL] GitCommitBear:
|    | Shortlog of HEAD commit isn't imperative mood, bad words are 'Removed'
```

see also coala.io/commit

**Comment by sils1297** [comment]
change it with amend ;)

**Comment by wimpykid26** [comment]
@sils1297 could you kindly review ? :)

File: coalib/__init__.py [hunk 7]
```diff
@@ -1,3 +1,11 @@
+"""
+The coalib package is a collection of various subpackages
+regarding writing and editing bears.
+Various other packages such as formatting and
+settings are also included in coalib.
```
Comment (sils1297) [inline coalib/__init__.py]: your way of breaking the line is pretty weird here, can't you break after 80 chars consistency? Just set it in your editor.

**Comment by sils1297** [comment]
Seems like you got the wrong commit message, you're adding the docs and not changing indentation right?

Also instead of `Fixes issue` just write `Fixes` because the issue won't be correctly closed otherwise automatically.

File: coalib/__init__.py [hunk 8]
```diff
@@ -1,3 +1,10 @@
+"""
+The coalib package is a collection of various subpackages regarding writing
+and editing bears. Various other packages such as formatting and settings are
```
Comment (sils1297) [inline coalib/__init__.py]: also executing, the main parts are about running bears and interacting with the user

Comment (sils1297) [inline coalib/__init__.py]: also executing, the main parts are about running bears and interacting with the user

**Comment by sils1297** [comment]
About your commit message again ;) :point_up: [September 21, 2016 5:22 PM](https://gitter.im/coala-analyzer/coala?at=57e2a5b2c3e7045a306a16b0) as described at coala.io/commit and also please use the full issue URL, not just the issue number so the link will also work when people look at it e.g. locally outside GitHub.

**Comment by wimpykid26** [comment]
@sils1297 Please review :)

**Comment by wimpykid26** [comment]
I added the full URL in the editor inside the git shell but it does not show in the commit message here. Even commit --amend shows the full url in the latest commit.

File: coalib/__init__.py [hunk 9]
```diff
@@ -1,3 +1,10 @@
+"""
```
Comment (Adrianzatreanu) [inline coalib/__init__.py]: commit message: "it's"->"its"

**Comment by sils1297** [comment]
yep github doesn't show it, you can see on mouseover wether it's the full
url or just the reference ;)

File: coalib/__init__.py (hunk 9 above)
Comment (Adrianzatreanu) [inline coalib/__init__.py]: you are wrapping the commit body weirdly in a way thats unreadable. please wrap at most at 80 chars, but not waaay less than 80 as you did. also end with period after "documentation"

**Comment by wimpykid26** [comment]
@sils1297 Please review :)

**Comment by sils1297** [comment]
ack https://github.com/coala/coala/pull/2773/commits/3448554d48423e5082cd86b3a1a9b708b147a7de

**Comment by sils1297** [comment]
@rultor merge

**Comment by sils1297** [comment]
Hey, this is your first contribution, right? Congrats and welcome to the coalaians! :)

Rultor will now automatically release your changes as a prerelease so users can experience the goodness of your changes right now.

See also https://github.com/coala-analyzer/coala/wiki/coala-Steam-Offer btw.

**Comment by wimpykid26** [comment]
@sils1297 Feels great to start things off !. Thank you for all the help that you provided. Would you recommend me to work on another newcomer issue or delve a little deep ? :)

**Comment by sils1297** [comment]
go for difficulty/low, if you see typos etc. please file newcomer issues as
we don't have enough ATM

Sincerely,

Lasse Schuirmann

lasse@schuirmann.net
http://coala-analyzer.org/ - http://viperdev.io/ - http://gitmate.com/

**Pull request merged**